src/language/printer.js

```diff
@@ -31,7 +31,8 @@ const printDocASTReducer = {
     const defs = wrap('(', join(node.variableDefinitions, ', '), ')');
     const directives = join(node.directives, ' ');
     const selectionSet = node.selectionSet;
-    return !name ? selectionSet :
+    return !name ? op === `query` ?
+      selectionSet : join([ op, selectionSet ], ' ') :
```

Comment:
unnamed queries can have variable defs and directives.

I think you may want to add a test case like:

```
mutation ($var: String) @directive { field }
```